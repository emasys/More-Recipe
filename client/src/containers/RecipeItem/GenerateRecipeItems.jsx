import React from 'react';
import Dropzone from 'react-dropzone';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Auth from '../../components/auth';

const GeneraterecipeItem = props => (
  <div>
    <div>
      <button
        className={`btn btn-success mb-5 ${props.state.save}`}
        onClick={props.handleImg}
      >
        Upload Image
      </button>
    </div>
    <figure
      className="img-wrapper"
      onMouseEnter={props.hoverIn}
      onMouseLeave={props.hoverOut}
    >
      {props.state.edit && (
        <div className={` changeDp hovered  ${props.state.status}`}>
          <Dropzone
            onDrop={props.handleDrop}
            accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png"
            multiple={false}
            className=" p-10 text-center text-light dropzone-dp"
          >
            click to upload / change picture
          </Dropzone>
        </div>
      )}

      <img
        src={props.state.preview || props.foodImg}
        alt="foodie"
        className="img-fluid rounded recipeImage"
      />
      {/* <figcaption className="text-center bolder">
        {props.state.recipeItem.recipe.name}
      </figcaption> */}
    </figure>
    <div className="d-inline mt-3 reaction-pane">
      <span className="text-center card-link" onClick={props.favIt}>
        <i
          className={`fa  ${
            props.state.favoriteStatus ?
              'fa-heart red animated bounceIn flash' :
              'fa-heart gray'
          } fa-2x`}
          aria-hidden="true"
          id="favorite"
        />
        <em className="bg-dark">{props.state.recipeItem.recipe.favorite}</em>
      </span>
      <span className="text-center card-link m-1" onClick={props.upvote}>
        <i
          className={`fa ${
            props.reactionUp.includes(Auth.userID()) ?
              'fa-thumbs-up animated bounceIn flash blue' :
              'fa-thumbs-up gray'
          } fa-2x`}
          aria-hidden="true"
          id="like"
        />
        <em className="bg-success">{props.state.recipeItem.recipe.upvote}</em>
      </span>
      <span className="text-center card-link m-1" onClick={props.downvote}>
        <i
          className={`fa ${
            props.reactionDown.includes(Auth.userID()) ?
              'fa-thumbs-down animated bounceIn flash red' :
              'fa-thumbs-down gray'
          } fa-2x`}
          aria-hidden="true"
          id="dislike"
        />
        <em className="bg-danger">{props.state.recipeItem.recipe.downvote}</em>
      </span>
      <span className="text-center card-link m-1" onClick={props.upvote}>
        <i
          className="fa fa-eye fa-2x"
          aria-hidden="true"
          id="views"
        />
        <em className=" bg-dark">{props.state.recipeItem.recipe.views}</em>
      </span>
      <span className="text-center card-link m-1" onClick={props.upvote}>
        <i
          className="fa fa-comments fa-2x"
          aria-hidden="true"
          id="views"
        />
        <em className=" bg-dark">{props.state.recipeItem.recipe.comments}</em>
      </span>
      <div className="m-1 float-right d-inline">
        <i className="fa fa-tag " aria-hidden="true" />
        <Link to={`/category/${props.state.recipeItem.recipe.category}`}>
          {props.state.recipeItem.recipe.category}
        </Link>
      </div>
    </div>
  </div>
);

GeneraterecipeItem.propTypes = {
  handleImg: PropTypes.func,
  state: PropTypes.object,
  hoverIn: PropTypes.func,
  hoverOut: PropTypes.func,
  handleDrop: PropTypes.func,
  favIt: PropTypes.func,
  upvote: PropTypes.func,
  downvote: PropTypes.func,
  foodImg: PropTypes.string
};
export default GeneraterecipeItem;
