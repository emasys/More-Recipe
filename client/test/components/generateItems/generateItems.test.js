import React from 'react';
import { mount } from 'enzyme';
import GenerateRecipeItems from '../../../src/containers/RecipeItem/GenerateRecipeItems';
import { singleRecipe } from '../../__mocks__/recipeMocks';

const props = {
  handleImg: jest.fn(),
  state: singleRecipe,
  reactionUp: [],
  reactionDown: [],
  hoverIn: jest.fn(),
  hoverOut: jest.fn(),
  handleDrop: jest.fn(),
  favIt: jest.fn(),
  upvote: jest.fn(),
  downvote: jest.fn(),
  foodImg: ''
};
jest.mock('react-router-dom');
test('should render favorite list correctly with one favorite recipe', () => {
  const wrapper = mount(<GenerateRecipeItems {...props} />);
  expect(wrapper.find('GeneraterecipeItem').length).toBe(1);
  expect(wrapper).toMatchSnapshot();
});

test('should render favorite', () => {
  const wrapper = mount(<GenerateRecipeItems {...props} />);
  const favBtn = wrapper.find('.fa-heart');
  favBtn.simulate('click');
  expect(favBtn.length).toBe(1);
  expect(props.favIt).toBeCalled();
  expect(wrapper).toMatchSnapshot();
});
