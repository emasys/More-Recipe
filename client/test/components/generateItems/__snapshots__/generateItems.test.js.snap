// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render favorite 1`] = `
<GeneraterecipeItem
  downvote={[Function]}
  favIt={[Function]}
  foodImg=""
  handleDrop={[Function]}
  handleImg={[Function]}
  hoverIn={[Function]}
  hoverOut={[Function]}
  reactionDown={Array []}
  reactionUp={Array []}
  state={
    Object {
      "recipeItem": Object {
        "recipe": Object {
          "category": "Breakfast",
          "comments": 1,
          "createdAt": "2017-12-31T18:42:38.431Z",
          "description": "axx",
          "direction": "xax",
          "downvote": 0,
          "favorite": 2,
          "foodImg": "http://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
          "id": 26,
          "ingredients": Array [
            "xs",
          ],
          "name": "msnbxmanmaxmnaxbaxbnbxasdadaasdad ba xbn xanb xaxasxxax",
          "reactionDown": Array [],
          "reactionUp": Array [],
          "recipeCount": null,
          "searchIng": null,
          "updatedAt": "2018-01-14T18:55:09.486Z",
          "upvote": 0,
          "userId": 1,
          "views": 11,
        },
      },
    }
  }
  upvote={[Function]}
>
  <div>
    <div>
      <button
        className="btn btn-success mb-5 undefined"
        onClick={[Function]}
      >
        Upload Image
      </button>
    </div>
    <figure
      className="img-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="foodie"
        className="img-fluid rounded recipeImage"
        src=""
      />
    </figure>
    <div
      className="d-inline mt-3 reaction-pane"
    >
      <span
        className="text-center card-link"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa  fa-heart gray fa-2x"
          id="favorite"
        />
        <em
          className="bg-dark"
        >
          2
        </em>
      </span>
      <span
        className="text-center card-link m-1"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-thumbs-up gray fa-2x"
          id="like"
        />
        <em
          className="bg-success"
        >
          0
        </em>
      </span>
      <span
        className="text-center card-link m-1"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-thumbs-down gray fa-2x"
          id="dislike"
        />
        <em
          className="bg-danger"
        >
          0
        </em>
      </span>
      <span
        className="text-center card-link m-1"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-eye fa-2x gray"
          id="views"
        />
        <em
          className=" bg-dark"
        >
          11
        </em>
      </span>
      <div
        className="m-1 float-right d-inline"
      >
        <i
          aria-hidden="true"
          className="fa fa-tag "
        />
        <Link
          replace={false}
          to="/category/Breakfast"
        />
      </div>
    </div>
  </div>
</GeneraterecipeItem>
`;

exports[`should render favorite list correctly with one favorite recipe 1`] = `
<GeneraterecipeItem
  downvote={[Function]}
  favIt={[Function]}
  foodImg=""
  handleDrop={[Function]}
  handleImg={[Function]}
  hoverIn={[Function]}
  hoverOut={[Function]}
  reactionDown={Array []}
  reactionUp={Array []}
  state={
    Object {
      "recipeItem": Object {
        "recipe": Object {
          "category": "Breakfast",
          "comments": 1,
          "createdAt": "2017-12-31T18:42:38.431Z",
          "description": "axx",
          "direction": "xax",
          "downvote": 0,
          "favorite": 2,
          "foodImg": "http://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
          "id": 26,
          "ingredients": Array [
            "xs",
          ],
          "name": "msnbxmanmaxmnaxbaxbnbxasdadaasdad ba xbn xanb xaxasxxax",
          "reactionDown": Array [],
          "reactionUp": Array [],
          "recipeCount": null,
          "searchIng": null,
          "updatedAt": "2018-01-14T18:55:09.486Z",
          "upvote": 0,
          "userId": 1,
          "views": 11,
        },
      },
    }
  }
  upvote={[Function]}
>
  <div>
    <div>
      <button
        className="btn btn-success mb-5 undefined"
        onClick={[Function]}
      >
        Upload Image
      </button>
    </div>
    <figure
      className="img-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="foodie"
        className="img-fluid rounded recipeImage"
        src=""
      />
    </figure>
    <div
      className="d-inline mt-3 reaction-pane"
    >
      <span
        className="text-center card-link"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa  fa-heart gray fa-2x"
          id="favorite"
        />
        <em
          className="bg-dark"
        >
          2
        </em>
      </span>
      <span
        className="text-center card-link m-1"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-thumbs-up gray fa-2x"
          id="like"
        />
        <em
          className="bg-success"
        >
          0
        </em>
      </span>
      <span
        className="text-center card-link m-1"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-thumbs-down gray fa-2x"
          id="dislike"
        />
        <em
          className="bg-danger"
        >
          0
        </em>
      </span>
      <span
        className="text-center card-link m-1"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-eye fa-2x gray"
          id="views"
        />
        <em
          className=" bg-dark"
        >
          11
        </em>
      </span>
      <div
        className="m-1 float-right d-inline"
      >
        <i
          aria-hidden="true"
          className="fa fa-tag "
        />
        <Link
          replace={false}
          to="/category/Breakfast"
        />
      </div>
    </div>
  </div>
</GeneraterecipeItem>
`;
